Sources/Metadata.swift

```diff
@@ -36,7 +36,9 @@ struct MetadataBuilder: Builder {
     let example: Any?
 
     init(map: Map) throws {
-        if let typeString: String = try? map.value("type"), let mappedType = DataType(rawValue: typeString) {
+        if map.JSON["$ref"] != nil {
+            type = .reference
+        } else if let typeString: String = try? map.value("type"), let mappedType = DataType(rawValue: typeString) {
```

Comment:
Mapping the data type should always be the first check. Only if the "type" was unset should we try to determine that from the other fields.